### PR TITLE
Feature/raspberry

### PR DESCRIPTION
**Pull request by p4p4**
please accept my pull request. This introduces RaspberryPi support.
please also merge into develop *as soon as possible*

- can be activated via build.gradle
- implements settings
- implements new Brick category
- implements 4 new Raspberry Bricks (SET pin, READ pin, PWM, when pin changed interrupt script)
- reading a pin value can also be used as a function in every Formula-Brick

Wiki Page: 
https://confluence.catrob.at/display/CATARDUINO/Raspberry+Pi+Integration

Video demo:
https://www.youtube.com/watch?v=8nuCalidgXo
https://www.youtube.com/watch?v=9UoarPCFj7Q

This is a RaspberryPi feature and has nothing to do with the Arduino feature, don't let yourself be fooled by the commit msg prefixes.

### CODE REVIEW

**Comment by p4p4** [comment]
I am reusing functionality of the BroadCast system, this is why I executed these tests. One of them failed, but it already fails in the current develop branch (-> not my fault)

executed test (Broadcasts)
BroadcastActionTest  [OK]:
https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-CustomEmulator-Single-ClassAndPackage-Test/675/

BroadCastReceiverRegressionTest [OK]:
https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-CustomEmulator-Single-ClassAndPackage-Test/677/

BroadcastBricksTest: [FAILS]
feature/raspberry result: https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-CustomEmulator-Single-ClassAndPackage-Test/676/
this is *not my fault*, since it also fails in the develop branch, see result of "develop":
develop branch result: https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-CustomEmulator-Single-ClassAndPackage-Test/678/

Error:
junit.framework.AssertionFailedError: View with id: '2131558475' is not found!
at com.robotium.solo.Solo.getView(Solo.java:2026)
at com.robotium.solo.Solo.getView(Solo.java:2006)
at org.catrobat.catroid.uitest.content.brick.BroadcastBricksTest.testBroadcastBricks(BroadcastBricksTest.java:112)
at android.test.InstrumentationTestCase.runMethod(InstrumentationTestCase.java:214)
at android.test.InstrumentationTestCase.runTest(InstrumentationTestCase.java:199)
at android.test.ActivityInstrumentationTestCase2.runTest(ActivityInstrumentationTestCase2.java:192)
at org.catrobat.catroid.uitest.util.BaseActivityInstrumentationTestCase.runBare(BaseActivityInstrumentationTestCase.java:137)
at android.test.AndroidTestRunner.runTest(AndroidTestRunner.java:191)
at android.test.AndroidTestRunner.runTest(AndroidTestRunner.java:176)
at android.test.InstrumentationTestRunner.onStart(InstrumentationTestRunner.java:555)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:1853)

**Comment by p4p4** [comment]
It's not my fault that the code is marked as unformatted:

the reported code was not written by me and already did not fulfill the coding standard before. See for example
https://github.com/Catrobat/Catroid/blame/develop/catroidTest/src/org/catrobat/catroid/test/io/StorageHandlerTest.java#L296

**Comment by robertpainsi** [comment]
> the reported code was not written by me

Hmm, that's odd :worried:
I'll check the formatter bot immediately :wrench:

**Comment by robertpainsi** [comment]
Here you go :smirk_cat:! Thanks for pointing it out instead of blindly adding the formatted commit. I like such an attitude a lot!

However, I still have no clue why it happend. :crying_cat_face: According to the log, the tests have been reformatted as well.
Unfortunately it's not reproducible :angry:. I just gave the bot another change to check your PR (and also did a manual check) and the output is correct now.

**Comment by p4p4** [comment]
Thank you! :smiley_cat: 
I am not sure about that behaviour either, maybe some of these wrong formatted Lego NXT code lines showed up during today's rebasing session. Then I probably performed the merge as intended, so maybe I introduced these wrong formatted LoC's during the rebase, even though I'm not the author of it.  But still, that is just one theory.

Nevertheless, the PR should (again) be ready now.

**Comment by robertpainsi** [comment]
Rebasing and other git commands shouldn't effect the bot since it works on hash basis. So don't worry, the problem is definitely on the bots side :wink: and you can continue doing all the dirty things you like with git :see_no_evil:

> Nevertheless, the PR should (again) be ready now.

I'll forward it to @thmq. Let me know if any code review is performed.

**Comment by avihric** [comment]
:warning: beware you are doing something wrong while rebasing, your commit history contains other commits that do not belong to this PR.
If you need assistance, there are some people in the panic room to help you out.

**Comment by p4p4** [comment]
Oh no! :angry:  Then the article should be updated.. (https://confluence.catrob.at/display/KNOWHOW/Git+Introduction)
all I was doing was
>To get your branch up to date with possible changes in the official repository ones 
>    git fetch upstream develop
>    git rebase upstream/develop

Today I can't go there unfortunately, but I probably will tomorrow.

**Comment by robertpainsi** [comment]
Just a second...

**Comment by robertpainsi** [comment]
@avihric
@p4p4 fixed it and the git history is clean now.

> Just a second...

commented 2 hours ago :joy_cat:

> Then the article should be updated..

Yeah, definitely! Between you and me, I've never liked these instructions. :scream_cat:

**Comment by p4p4** [comment]
note: code review is performed next Monday (February 8th) with @thmq . If anybody wants to help/point out improvements, feel free to join us or comment here ;)

File: catroid/res/layout/brick_raspi_pwm.xml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            android:id="@+id/brick_raspi_pwm_percentage_edit_text"
+            style="@style/BrickEditText"
+            app:layout_inputType="number"
+            app:layout_textField="true" />
+
+        <TextView
+            android:id="@+id/brick_raspi_pwm_percent_character"
+            style="@style/BrickText.SingleLine"
+            app:layout_newLine="true"
+            android:text="%" />
```
Comment (p4p4) [inline catroid/res/layout/brick_raspi_pwm.xml]: is this allowed?

**Pull request merged**